### PR TITLE
Add slug section to API docs

### LINKED ISSUE #15: Define slug regex
Slugs appear in many places in TimeSync. They are used to get objects from the
backend, reference objects from within other objects, etc. Define their spec.

### PR DESCRIPTION
**Pull request by tschuy**
Fixes #15.

### CODE REVIEW

**Comment by pop** [comment]
the regex description is good and the example looks valid; according to some tests with http://regexpal.com/ is matches our unique slug definition. +1

File: source/draft_api.rst [hunk 1]
```diff
@@ -38,6 +38,23 @@ For example: https://timesync.osuosl.org/v1/projects
...
+backend, reference objects from within other objects, etc. A valid slug follows
+a very specific format:
+
+#) May only contain lowercase letters and numbers
+#) Sets of lowercase letters and numbers can be separated with a single hyphen
+#) Must begin and end with an alphanumeric character
+
+A regular expression matching a valid slug is::
+
+    ^(?:[0-9]*-)*[0-9]*[a-z]+[a-z0-9]*(?:-[a-z0-9]+)*$
```
Comment (Kennric) [inline source/draft_api.rst]: Lets leave out the regex for now, and just state the specs.

**Comment by tschuy** [comment]
@Kennric forgot about this PR. I updated the docs to match what our discussions ended with and what timesync-node has.

**Comment by Kennric** [comment]
Looks good to me. +1

**Pull request merged**